### PR TITLE
Godoc examples

### PR DESCRIPTION
**Pull request by svekars**
- Added Go API client example for:
   - Creating a repo
   - Listing a repo
   - Deleting a repo
   - Putting a File into a repo

### CODE REVIEW

**Comment by svekars** [comment]
The user will not need to access the godoc_test.go file. The doc build instructions are for contributors only. When this is merged, the content should automatically appear in our godoc reference - https://godoc.org/github.com/pachyderm/pachyderm. You can preview locally by setting up the server on your machine according to the instructions. But the users won't need to do it. 

Here is a couple of screenshots of how it will look locally:

Screen Shot 2020-04-02 at 4 55 26 PM
Screen Shot 2020-04-02 at 4 55 39 PM
Screen Shot 2020-04-02 at 4 55 33 PM

I can probably add a note about how to execute Go example and a sample go program here:  https://docs.pachyderm.com/latest/reference/clients/#go-client

File: doc/docs/master/reference/clients.md [hunk 1]
```diff
@@ -25,6 +25,19 @@ cd $GOPATH/src/google.golang.org/grpc
 git checkout 21f8ed309495401e6fd79b3a9fd549582aed1b4c
 ```
 
+### Running Go Examples
+
+The Pachyder, [godocs](https://godoc.org/github.com/pachyderm/pachyderm/src/client) reference
```
Comment (pappasilenus) [inline doc/docs/master/reference/clients.md:30]: fix typo:

The Pachyderm, [godocs](https://godoc.org/github.com/pachyderm/pachyderm/src/client) reference

File: examples/opencv/goclient-example/opencv-example.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+	if _, err := c.PutFile("images", "master", "AT-AT.png", file2); err != nil {
+		panic(err)
+	}
+
+	file3, err := os.Open("examples/opencv/kitten.png")
+	if err != nil {
+		panic(err)
+	}
+
+	if _, err := c.PutFile("images", "master", "kitten.png", file3); err != nil {
```
Comment (pappasilenus) [inline examples/opencv/goclient-example/opencv-example.go:60]: I don't think  it's productive to include the images in the repo.  Should you use a call to PutFileURL, instead?

File: examples/opencv/goclient-example/opencv-example.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+	if _, err := c.PutFile("images", "master", "liberty.png", file1); err != nil {
+		panic(err)
+	}
+
+	file2, err := os.Open("examples/opencv/AT-AT.png")
+	if err != nil {
+		panic(err)
+	}
+
+	if _, err := c.PutFile("images", "master", "AT-AT.png", file2); err != nil {
```
Comment (pappasilenus) [inline examples/opencv/goclient-example/opencv-example.go:51]: see comment on kitten, below

File: examples/opencv/goclient-example/opencv-example.go [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+	file1, err := os.Open("examples/opencv/liberty.png")
+	if err != nil {
+		panic(err)
+	}
+
+	if _, err := c.PutFile("images", "master", "liberty.png", file1); err != nil {
+		panic(err)
+	}
+
+	file2, err := os.Open("examples/opencv/AT-AT.png")
```
Comment (pappasilenus) [inline examples/opencv/goclient-example/opencv-example.go:46]: see comment on kitten, below

File: examples/opencv/goclient-example/README.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+To run this example, you must have the [Go](https://golang.org/)
+programming language installed and configured on your computer.
+
+To run the OpenCV example, execute `opencv-example.go` from the
+root of the Pachyderm repository.
+
+**Example:**
+
+```
+go run go run examples/opencv/goclient-example/opencv-example.go
```
Comment (pappasilenus) [inline examples/opencv/goclient-example/README.md:13]: remove the first "go run",  I think it's a typo.

File: examples/opencv/goclient-example/opencv-example.go [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+	"github.com/pachyderm/pachyderm/src/client/pps"
+	//	"os"
+)
+
+func main() {
+
+	// Replace the IP address with your `pachd` address.
+	// If running in minikube, this will be your minikube
+	// IP.
+	c, err := client.NewFromAddress("192.168.64.2:30650")
```
Comment (pappasilenus) [inline examples/opencv/goclient-example/opencv-example.go:21]: change this to "localhost:30650" and it works out-of-box.

**Pull request closed without merging**